app/src/main/java/com/example/hackernews/data/CallApi.kt

```diff
@@ -1,34 +1,73 @@
 package com.example.hackernews.data
 
 import android.content.Context
-import android.util.Log
-import android.widget.Toast
-import com.example.hackernews.comments.CommentsAdapter
+import com.example.hackernews.callbacks.LoadDataCallback
 import com.example.hackernews.constants.Api
 import com.example.hackernews.helpers.Helper
-import com.example.hackernews.models.Comment
 import com.example.hackernews.models.NewsM
-import com.example.hackernews.news.NewsAdapter
 import com.example.hackernews.news.NewsDataType
 import com.example.hackernews.services.NewsService
-import retrofit2.Call
-import retrofit2.Callback
-import retrofit2.Response
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
 import java.io.Serializable
+import java.lang.Exception
 
 class CallApi(val context: Context) : Serializable{
 
-    private val retrofit: Retrofit by lazy {
+    private val retrofit: NewsService by lazy {
         Retrofit.Builder()
                 .baseUrl(Api.BASE_URL)
                 .addConverterFactory(GsonConverterFactory.create())
-                .build()
+                .build().create(NewsService::class.java)
     }
 
-    private val newsApi by lazy {
-        retrofit.create(NewsService::class.java)
+    // this code works, but it is pretty slow, 10 news per sec
+    // try to find causer
+    fun getStories(newsDataType: NewsDataType, loadData: LoadDataCallback) {
+        CoroutineScope(Dispatchers.IO).launch {
+           val storiesIds = retrofit.getStoriesIds(newsDataType.rawValue).body()
+            val allNews = ArrayList<NewsM>()
+            val iterator = (0..storiesIds!!.size).iterator()
```

Comment:
The use of iterators in Kotlin is very rare like never 😄 

We have a ton of built-in functions to handle different processing on lists e.g. `forEach`, `map,` etc.